### PR TITLE
Ensure builds containing deployements report accurate build time

### PR DESCRIPTION
**Pull request by lacasseio**
### Context

related https://github.com/gradle/gradle/issues/6789

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Fissue-6789)
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        then:
+        assertDeploymentIsRunning(key)
+        lastBuildTime >= one(buildTimes)
+    }
+
+    List<Integer> getBuildTimes() {
+        return (output =~ /BUILD SUCCESSFUL in (\d+)s/).collect { it[1].toString().toInteger() }
+    }
+
+    private static <T> T one(Iterable<T> iterable) {
```
Comment (big-guy) [inline subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy]: I think this is the same as `CollectionUtils.single`

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ */
+
+package org.gradle.launcher.continuous
+
+class DeploymentContinuousBuildIntegrationTest extends Java7RequiringContinuousIntegrationTest {
+    File triggerFile = file('triggerFile')
+    File keyFile = file('keyFile')
+
+    def setup() {
+        buildFile << """
```
Comment (big-guy) [inline subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy]: Can you pull this into a test fixture to share this with the other test? This looks a lot like DeploymentHandleContinuousBuildCrossVersionSpec's code?

The trigger/key files above would be part of the fixture too and the assert method below.

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy (hunk 1 above)
Comment (lacasseio) [reply subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy]: Done.

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy (hunk 2 above)
Comment (lacasseio) [reply subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy]: Done.

**Pull request merged**